adventure/adventure.py

```diff
@@ -2925,14 +2932,39 @@ def E(t: str) -> str:
             except Exception:
                 log.error("Error with the new character sheet", exc_info=True)
                 continue
-            bonus_tinkerer = c.att + c.skill["att"]
             if c.heroclass["name"] == "Tinkerer" and not sharpen:
+                bonus_tinkerer = c.att + c.skill["att"]
                 sharpen_chance = min(int(bonus_tinkerer / 2.5 + 1), c.lvl)
                 sharpen_roll = random.randint(1, 100)
                 if sharpen_roll in range (1, sharpen_chance):
                     sharpen = True
                     sharpen_bonus = int(sharpen_chance * 0.4) + 3
                     msg += f"{bold(self.E(user.display_name))} sharpened the weapons of the party! *[🗡 +{sharpen_bonus}%]*\n"
+            if c.heroclass["name"] == "Bard" and not incision:
+                bonus_bard = int((c.att + c.skill["att"] + c.cha + c.skill["cha"]) / 2)
+                incision_chance = min(int(bonus_bard / 2.5 + 1), c.lvl)
+                incision_roll = random.randint(1, 100)
+                if incision_roll in range (1, incision_chance):
+                    incision = True
+                    incision_bonus = int(incision_chance * 0.4) + 3
+                    msg += (
+                        f"{bold(self.E(user.display_name))} strikes precise incisions with his dagger! "
+                        f"*[-{incision_bonus}% to physical resistance]*\n"
+                    )
+        for user in session.magic: #check if a tinkerer is in the magic party and calculate the possible bonus
+            try:
+                c = await Character._from_json(self.config, user)
+            except Exception:
+                log.error("Error with the new character sheet", exc_info=True)
+                continue
+            bonus_tinkerer = c.int + c.skill["int"]
+            if c.heroclass["name"] == "Tinkerer" and not potion:
```

Comment:
pedantic perhaps, but for each of these we could check "potion" first before fetching character. 
e.g. If potion is set there is no need to continue and fetch each character